src/main/java/com/telekom/javaschool/medicalrehapp/service/DoctorServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.telekom.javaschool.medicalrehapp.service;
+
+import com.telekom.javaschool.medicalrehapp.dao.DoctorRepository;
+import com.telekom.javaschool.medicalrehapp.dto.DoctorDto;
+import com.telekom.javaschool.medicalrehapp.mapper.DoctorMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class DoctorServiceImpl implements DoctorService {
+
+    private final DoctorRepository doctorRepository;
+    private final DoctorMapper doctorMapper;
+
+    @Autowired
+    public DoctorServiceImpl(DoctorRepository doctorRepository, DoctorMapper doctorMapper) {
+        this.doctorRepository = doctorRepository;
+        this.doctorMapper = doctorMapper;
+    }
+
+    @Transactional
+    @Override
+    public void save(DoctorDto doctorDto) {
+        doctorRepository.save(doctorMapper.dtoToEntity(doctorDto));
+    }
+
+    @Transactional
```

Comment:
1. I think you wanted to use org.springframework.transaction.annotation.Transactional rather than javax.transaction.Transactional
2. For read methods it's better to use @Transactional(readOnly = true)

Remember the DB+JPA lecture